lib/ansible/plugins/connection/__init__.py

```diff
@@ -307,6 +307,17 @@ def close(self):
         if self._connected:
             self._connected = False
 
+    def get_options(self, hostvars=None):
+        options = super(NetworkConnectionBase, self).get_options(hostvars=hostvars)
+
+        if self._sub_plugin.get('obj') and self._sub_plugin.get('type') != 'external':
```

Comment:
If `self._sub_plugin.get('obj')` of type list can allow registering multiple sub_plugins. The network_cli connection plugin currently adds `cliconf` as subplugin. If it is list it can also add plugin options for `ssh_type` plugin (paramiko_ssh/libssh).

Reply:
We can change it so sub_plugin is a list and deal with all the plugins at once here. The issue comes with doing that in a way that won't break compatibility with older versions of ansible.

It is definitely a thing we can and probably should do, but it will require more coordination with netcommon to be done properly. In the meantime, this PR loses no functionality and the paramiko/libssh issues can be handled by https://github.com/ansible-collections/ansible.netcommon/pull/271

Reply:
Compatibility goes both ways, so if changed to a list i would still fallback to it's current form (optionally add deprecation notice to this path).  

As for making the the collection code compatible, that is a bit harder as it would require a wrapper object that directs to proper data structure depending on 'callable'.